internal/registry/types.bucket_test.go

```diff
@@ -192,5 +209,123 @@ func TestBucket_UpdateLabelsForBuild_withMultipleBuilds(t *testing.T) {
 	if ok := cmp.Equal(registeredBuilds[0].Labels, registeredBuilds[1].Labels); ok {
 		t.Errorf("expected registered builds to have different labels but they are equal")
 	}
+}
+
+func TestBucket_PopulateIteration(t *testing.T) {
+	tc := []struct {
+		desc              string
+		buildName         string
+		bucketBuildLabels map[string]string
+		buildLabels       map[string]string
+		labelsCount       int
+		buildCompleted    bool
+		noDiffExpected    bool
+	}{
+		{
+			desc:      "existing incomplete build",
+			buildName: "happcloud.image",
+			bucketBuildLabels: map[string]string{
+				"version":   "1.7.0",
+				"based_off": "alpine",
+			},
+			labelsCount:    2,
+			buildCompleted: false,
+			noDiffExpected: true,
+		},
+		{
+			desc:      "existing incomplete build with improperly set build labels",
+			buildName: "happcloud.image",
+			bucketBuildLabels: map[string]string{
+				"version":   "1.7.3",
+				"based_off": "alpine-3.14",
+			},
+			buildLabels: map[string]string{
+				"version":   "packer.version",
+				"based_off": "var.distro",
+			},
+			labelsCount:    2,
+			buildCompleted: false,
+			noDiffExpected: true,
+		},
+		{
+			desc:      "completed build with no labels",
+			buildName: "happcloud.image",
+			bucketBuildLabels: map[string]string{
+				"version":   "1.7.0",
+				"based_off": "alpine",
+			},
+			labelsCount:    0,
+			buildCompleted: true,
+			noDiffExpected: false,
+		},
+		{
+			desc:      "existing incomplete build with extra previously set build label",
```

Comment:
Let me see if I got it right, if I rerun an incomplete iteration, can I add more build labels the second time I try building it?

Reply:
No wait, is it the other way around? If I'm trying to complete an existing iteration, I can't update labels?

Reply:
It is actually the first one, but the example is a little confusing. Right now if a user sets one build label and the build fails for whatever reason the build label will get persisted. Now lets say when they rerun the failed build they decide to add a few extra build labels the new labels will get added but the old label will not get removed. 

If the value for the previous label is modified the old persisted label will get updated to reflect the new value.

Reply:
Ahh got it! I understood that aftewards. Makes sense!